### ISSUE TITLE
No "update" function from TCP remote control

### ISSUE THREAD

**Issue by lokinou** [labstreaminglayer/App-LabRecorder#57]
Hello, 
LabRecorder is going to be helpful in setting up an EEG experiment in my lab. Thank you for the great work !

I am remotely controlling labrecorder and fail to send an update command. I have to restart labrecorder which is not optimal in my case.

I am not very good in compiling c++ so I have laid out the code that I believe would work to unexpensively add such a function:
in tcpinterface.cpp
```
void RemoteControlSocket::handleLine(QString s, QTcpSocket *sock) {
	qInfo() << s;
	if (s == "start")
		emit start();
	else if (s == "update")
		emit refresh_streams();
		// todo: perhaps add a 1sec delay to match with lsl stream scan timeout duration
```
in tcpinterface.h
```
signals:
	void refresh_streams();
	void start();
	void stop();
	void filename(QString s);
	void select_all();
	void select_none();
```

in mainwindow.cpp
```
void MainWindow::enableRcs(bool bEnable) {
	[...]
		connect(rcs.get(), &RemoteControlSocket::start, this, &MainWindow::rcsStartRecording);
		connect(rcs.get(), &RemoteControlSocket::refresh_streams, this, &MainWindow::rcsRefreshStreams);
[...]

void MainWindow::rcsRefreshStreams() {
	// update streams from rcs
	
	hideWarnings = true;
	refreshStreams();
}
```

in mainwindow.h
```
private slots:
	[...]
	void rcsStartRecording();
	void rcsRefreshStreams();
```
Hope it helps.
Cheers !

**Comment by AdoNunes** [comment]
Indeed it would be a great addition to a great tool. Out of the 5 buttons in the GUI, the update is the only one without a TCP equivalent command.
If the GUI is started before the creation of the streams, even after sending a select all, the new streams won't be detected. This greatly limits the functionality of remote control.

**Comment by tstenner** [comment]
Good catch and thanks for already writing the code :-). The change to `hideWarnings` wasn't needed, so the changes (#59) ended up a bit shorter.

**Comment by cboulay** [comment]
Sorry for crossing streams but I just updated the README and added a new tag.

**Comment by cboulay** [comment]
By the way, for those who are following this issue, depending on your exact use case there may already have been a way to achieve what you want and it might even have worked better.

If you use the [Required Streams](https://github.com/labstreaminglayer/App-LabRecorder/blob/master/LabRecorder.cfg#L42-L48) feature in the config file then you can specify streams that might be initially considered `missingStreams` then will be passed into the recording thread in the list of `watchfor` streams: https://github.com/labstreaminglayer/App-LabRecorder/blob/master/src/recording.h#L60

These streams will each get their own query and thread that looks for them: https://github.com/labstreaminglayer/App-LabRecorder/blob/master/src/recording.cpp#L124-L159

So, long story short, if you already know the name and hostname of the origin streams that you want to record, you can add these to your .cfg file and you can start recording even before they're online, then once those streams come online they will be automatically added to the xdf file.

**Comment by lokinou** [comment]
Thanks @tstenner for the pull request ! Now the list of RCS supported functions looks more complete.
@cboulay I actually used the `watchfor` functionnality  you mentionned.
I assumed undetected streams were marked as red to remind the user that those were missing, not that they would be caught later on, which is actually quite a nice desing!!!
I don't want to spam issues, but considering thus functionnality, I'm wondering if an `onlyrecord-watchfor` functionnality would help. I am on a university network and thus I get random lsl streams from everywhere :)

Thanks again!

**Comment by tstenner** [comment]
Then I've got good news for you: the `onlyrecord-watchfor` is the only/default mode of operation :-)

The list entries have three states:

1. currently found (green) but not checked -> won't be recorded
2. currently found (green) and checked  ->  will be recorded
3. currently not found (red), but checked  -> will be recorded as soon as it's found
